airbyte-cdk/python/unit_tests/sources/declarative/partition_routers/test_list_partition_router.py

```diff
@@ -92,6 +92,21 @@ def test_request_option(request_option, expected_req_params, expected_headers, e
     assert expected_body_json == partition_router.get_request_body_json(stream_slice=stream_slice)
```

Comment:
_This comment is for the file, not this specific line_

It is nice to see that no tests are breaking hence this is backward compatible but should we add test to highlight the new behavior that was added?

Reply:
@maxi297 which new behavior are you referring to? The list partition router shouldn't behave differently https://github.com/airbytehq/airbyte/pull/35471/files#diff-896828e962e8284dbb3edf6fca03651534337b3bf5043e645c3913703ab0584bL8-L87

Reply:
The return type of `stream_slices` is now `PerPartitionStreamSlice`. I don't see that reflected anywhere